src/System.Linq/tests/UnionTests.cs

```diff
@@ -208,6 +295,15 @@ public void ForcedToEnumeratorDoesntEnumerate()
         }
 
         [Fact]
+        public void ForcedToEnumeratorDoesntEnumerateMultipleUnions()
+        {
+            var iterator = NumberRangeGuaranteedNotCollectionType(0, 3).Union(Enumerable.Range(0, 3)).Union(Enumerable.Range(2, 4)).Union(new[] { 9, 2, 4 });
+            // Don't insist on this behaviour, but check its correct if it happens
+            var en = iterator as IEnumerator<int>;
+            Assert.False(en != null && en.MoveNext());
```

Comment:
That you can't just suddenly start enumerating through what was given to you as an enumerable, possibly messing up "proper" use later. After briefly having a "clever idea" I soon realised was a very silly idea, I added checks on this for every iterator type, both hand-written and compiler-generated, and have kept it updated since. It'll complain to the over-enthusiastic tweaker.